### DISCUSSION TITLE
where is graal version is coming from

### DISCUSSION THREAD

**Discussion by ozkanpakdil** [sdkman/sdkman-cli#1305] [Q&A] [answered]
I needed to install Graal 22.3 when I went to their [download page](https://www.graalvm.org/downloads/) shows
```
sdk install java 22.0.2-graal
```
![image](https://github.com/user-attachments/assets/216bc7e0-b1ec-4b8b-86de-c1d8ad87f0bd)

but my project was telling me 
```
[ERROR] Failed to execute goal org.graalvm.buildtools:native-maven-plugin:0.9.19:compile (default-cli) on project nacos-config-example
: Execution default-cli of goal org.graalvm.buildtools:native-maven-plugin:0.9.19:compile failed: GraalVM version 22.3 is required but
 22.1 has been detected, please upgrade. -> [Help 1]
```

Then I went and found [this post](https://github.com/oracle/graal/issues/6128#issuecomment-1457731233) I check `sdk list java` and could not see 22.3 graal package there too, when I ran the command from the post `sdk install java 22.3.1.r17-grl` , sdkman started downloading, and I can see the folder in my system, but I am not sure where is that downloaded from.
Now I have 3 questions
1. Why graal version 22.3 is not visible in `sdk java list` ?
2. Where is the package downloaded from ?
3. Are there any web page where I can see the latest graal versions ready for sdkman to install ?
Thank you

**Comment by helpermethod** [comment, marked as answer]
Hi @ozkanpakdil!

Do you need GraalVM 22.3 or just the lastest version of GraalVM? There was a naming change a while ago:

https://www.graalvm.org/release-calendar/#graalvm-enterprise-and-community-editions

**TLDR;** At some point GraalVM releases were based on the calendar year, nowadays they are based on the JDK numbering scheme.

22.3 e.g. was released October 18, 2022.

So, if you just need the latest version, based on JDK 22, use

    sdk install java 22.0.2-graal

for Oracle GraalVM (not Open Source) or

    sdk install java 22.0.2-graalce

for the GraalVM Community Edition (Open Source).

If you need the latest GraalVM version based on JDK 17, install `17.0.9` or `17.0.9-graalce` respectively.

Hope that helps!

**Reply by ozkanpakdil** [reply]
Hi @helpermethod 
Thank you for your answer.  Looks 22.3 is released at Jul 25 2023
![image](https://github.com/user-attachments/assets/f4403e35-1c60-4368-b6ab-0e39cbffd9d4)
Sound awkward, :) 22.0.2 is newer than 22.3 😄 mathematics just left the building. Java version numbers over the years has been funny anyway. 
1.6
1.7
8
11
16 <- google was giving 1.6 answers for any java 16 search and same for 1.5 vs 15
17 <- for a long time whenever I search anything on google I was getting java 1.7 related links
now 22 but we never know they may think to jump to 160 that way we will search and find java 1.6 docs again
